test/TestProjects/ProtocolMethodsInRestClient/Generated/TestServiceRestClient.cs

```diff
@@ -0,0 +1,404 @@
+        {
+            if (resourceId == null)
+            {
+                throw new ArgumentNullException(nameof(resourceId));
+            }
+
+            using var message = CreateDeleteRequest(resourceId);
+            _pipeline.Send(message, cancellationToken);
+            switch (message.Response.Status)
+            {
+                case 204:
+                    return message.Response;
+                default:
+                    throw ClientDiagnostics.CreateRequestFailedException(message.Response);
+            }
+        }
+
+        internal HttpMessage CreateDeleteRequest(string resourceId, RequestContext context)
+        {
+            var message = _pipeline.CreateMessage(context);
+            var request = message.Request;
+            request.Method = RequestMethod.Delete;
+            var uri = new RawRequestUriBuilder();
+            uri.Reset(_endpoint);
+            uri.AppendPath("/template/resources/", false);
+            uri.AppendPath(resourceId, true);
+            request.Uri = uri;
+            message.ResponseClassifier = ResponseClassifier204.Instance;
+            return message;
+        }
+
+        /// <summary> Delete resource. </summary>
+        /// <param name="resourceId"> The id of the resource. </param>
+        /// <param name="context"> The request context, which can override default behaviors on the request on a per-call basis. </param>
+        /// <exception cref="ArgumentNullException"> <paramref name="resourceId"/> is null. </exception>
+        /// <exception cref="ArgumentException"> <paramref name="resourceId"/> is an empty string, and was expected to be non-empty. </exception>
+#pragma warning disable AZC0002
+        public virtual async Task<Response> DeleteAsync(string resourceId, RequestContext context = null)
+#pragma warning restore AZC0002
+        {
+            Argument.AssertNotNullOrEmpty(resourceId, nameof(resourceId));
+
+            using var scope = ClientDiagnostics.CreateScope("TestServiceClient.Delete");
+            scope.Start();
+            try
+            {
+                using HttpMessage message = CreateDeleteRequest(resourceId, context);
+                return await _pipeline.ProcessMessageAsync(message, context).ConfigureAwait(false);
+            }
+            catch (Exception e)
+            {
+                scope.Failed(e);
+                throw;
+            }
+        }
+
+        /// <summary> Delete resource. </summary>
+        /// <param name="resourceId"> The id of the resource. </param>
+        /// <param name="context"> The request context, which can override default behaviors on the request on a per-call basis. </param>
+        /// <exception cref="ArgumentNullException"> <paramref name="resourceId"/> is null. </exception>
+        /// <exception cref="ArgumentException"> <paramref name="resourceId"/> is an empty string, and was expected to be non-empty. </exception>
+#pragma warning disable AZC0002
```

Comment:
Curious: do we still need these warning disables?

Reply:
I just shared the same code we generate for DPG. Please let me know if we want to remove this warning for RestClient.

Reply:
just want to clarify, are you thinking​ to remove the warning for all the DPG libraries or just for RestClient?

Reply:
This warning is for:
`"DO ensure all service methods, both asynchronous and synchronous,  take an optional CancellationToken parameter called cancellationToken.", "Client method should have cancellationToken as the last optional parameter (both name and it being optional matters"` ([here](https://github.com/Azure/azure-sdk-tools/blob/main/src/dotnet/Azure.ClientSdk.Analyzers/Azure.ClientSdk.Analyzers/Descriptors.cs#L17-L23))

So I don't think we should remove this warning because we don't take CancellationToken parameter for protocol methods.

Reply:
@KrzysztofCwalina told me he had turned off this analyzer, but I haven't looked into it yet.  Would you be willing to look in the azure-sdk-tools repo to see if this happened?

Reply:
Seems like we have turned off this analyzer because the code is compiling even if I remove `#pragma warning disable AZC0002`. I tested in [DeviceUpdate](https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/deviceupdate/Azure.IoT.DeviceUpdate/src/Generated/DeviceUpdateClient.cs#L122) in azure-sdk-for-net repo.

Reply:
@annelo-msft This PR will create a lot of noise if I do this change. Hence, I created a separate PR to remove this https://github.com/Azure/autorest.csharp/pull/1993. Could you please take a look?

Reply:
Approved!